snap/README.md

```diff
@@ -0,0 +1,172 @@
+edgexfoundry.security-secretstore-setup    enabled   inactive  -
+edgexfoundry.support-notifications         disabled  inactive  -
+edgexfoundry.support-scheduler             disabled  inactive  -
+edgexfoundry.sys-mgmt-agent                disabled  inactive  -
+edgexfoundry.vault                         enabled   active    -
+```
+
+##### Configure edgexfoundry
+
+Enable scheduler, to use Interval and interval actions for cleaning up Redis database periodically:
+
+```bash
+$ sudo snap set edgexfoundry support-scheduler=on 
+$ sudo snap set edgexfoundry support-notifications=on 
+```
+
+Enable system management agent service for communication between SMA and core data, to use `system services monitor` for system health check, metric check, configuration and operation:
+
+```bash
+$ sudo snap set edgexfoundry sys-mgmt-agent=on
+```
+
+Enable rules engine, to create streams and sinks:
+
+```bash
+$ sudo snap set edgexfoundry kuiper=on
+```
+
+Learn more of using edgexfoundry snap, please see [here](https://github.com/edgexfoundry/edgex-go/blob/main/snap/README.md)
+
+### JWT token
+
+API Gateway is needed to enter UI secure mode. Before the API Gateway can be used, a user and group must be created, and a JWT access token generated. This can be accomplished via the `secrets-config` command, or by using `snap set` commands
+
+The first step is to add a user. You need to create a public/private keypair, which can be done with:
+
+```
+$ openssl ecparam -genkey -name prime256v1 -noout -out private.pem
+$ openssl ec -in private.pem -pubout -out public.pem
+```
+
+The second step is if you then create the user using the secrets-config command, then you need to provide:
+
+- The username
+- The public key
+- (optionally) ID. This is a unique string identifying the credential. It will be required in the next step to create the JWT token. If you don't specify it, then an auto-generated one will be output by the secrets-config command
+
+Do this using `snap set`:
+
+```bash
+$ sudo snap set edgexfoundry env.security-proxy.user=user01,USER_ID,ES256
+$ sudo snap set edgexfoundry env.security-proxy.public-key="$(cat public.pem)"
+```
+
+The third step is then to generate a token using the user ID which you specified:
+
+```bash
+$ edgexfoundry.secrets-config proxy jwt --algorithm ES256 --private_key private.pem --id USER_ID --expiration=1h
+```
+
+An output has a long alphanumeric sequence of the format < alphanumeric >.< alphanumeric >.< alphanumeric >:
+
+```
+eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJVU0VSX0lEIiwiZXhwIjoxNjM1OTM1NjYxLCJuYmYiOjE2MzU5MzIwNjEsImlhdCI6MTYzNTkzMjA2MX0.Kzsu_3L7wVNbjAbGfNatmyVtsmDaodOG_b5nCHKywU327xdfqpgno1g3ai1yu8hz6pG0_2eHnordCSWxaPFoMA
+```
+
+This output is the JWT token for login UI secure mode. This token will be asked again if you exit the web page, and re-enter it. Please write the token you generated down in case you re-enter the UI web page. The needed token can not be generated for the second time
```

Comment:
Proposing change to: This output is the JWT token for UI login in secure mode. Please keep the token in a safe place for future re-use as the same token cannot be regenerated or recovered from EdgeX's secret-config CLI. The token is required each time you reopen the web page.

Reply:
Updated.